
See https://jira.mariadb.org/browse/MDEV-21595 for details.
Hopefully this will be fixed in a future release (it also affects AIX)

diff -wpruN '--exclude=*.orig' a~/storage/innobase/include/rem0types.h a/storage/innobase/include/rem0types.h
--- a~/storage/innobase/include/rem0types.h	1970-01-01 00:00:00
+++ a/storage/innobase/include/rem0types.h	1970-01-01 00:00:00
@@ -31,7 +31,8 @@ Created 5/30/1994 Heikki Tuuri
 typedef byte	rec_t;
 
 /** This type represents a field offset in a rec_t* */
-typedef unsigned short int offset_t;
+typedef unsigned short int rem0offset_t;
+#define offset_t rem0offset_t
 
 /* Maximum values for various fields (for non-blob tuples) */
 #define REC_MAX_N_FIELDS	(1024 - 1)
